Delete grandfathered Caffe2 dispatch keys.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56939 Delete grandfathered Caffe2 dispatch keys.**

These never have kernels registered to them and are effectively useless.
What I am not so sure if we allocate tensors to them or not; if we do
I cannot use asserts and I need to ensure we just return undefined
or something equivalent.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D28006160](https://our.internmc.facebook.com/intern/diff/D28006160)

ezyang committed:
Delete grandfathered Caffe2 dispatch keys.

These never have kernels registered to them and are effectively useless.
What I am not so sure if we allocate tensors to them or not; if we do
I cannot use asserts and I need to ensure we just return undefined
or something equivalent.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ailzhang reviewed (approved):
Thanks @ezyang !

bhosmer reviewed (approved):
If the worry is that there are tensors being allocated to these devices in fbcode that won't be caught in CI, maybe a better assert message?

ezyang commented:
@bhosmer yeah yeah, let me do that

ezyang committed:
Update on "Delete grandfathered Caffe2 dispatch keys."


These never have kernels registered to them and are effectively useless.
What I am not so sure if we allocate tensors to them or not; if we do
I cannot use asserts and I need to ensure we just return undefined
or something equivalent.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D28006160](https://our.internmc.facebook.com/intern/diff/D28006160)

[ghstack-poisoned]
